libs/ml/include/ml/ops/weights.hpp

```diff
@@ -210,6 +211,40 @@ class Weights : public fetch::ml::ops::PlaceHolder<T>, public Trainable<T>
     }
   }
 
+  /**
+   * interface to call standard weights initialisation routines. defaults to xavier.
+   * Fan in and fan out xavier not permitted with input and output sizes not known independently
+   * @param mode  An enum indicating which type of initialisation to perform
+   */
+  static void Initialise(ArrayType &array, std::uint64_t data_size,
+                         WeightsInitialisation mode = WeightsInitialisation::XAVIER_GLOROT,
+                         SizeType              seed = 123456789)
+  {
+    switch (mode)
+    {
+    case WeightsInitialisation::ONES:
+    {
+      array.Fill(1);
```

Comment:
Are you sure that it will work for any DataType?
FIll(static_cast<DataType>(1))